### ISSUE TITLE
EMAXBUFFER: handle upload canceling / content-type checks

### ISSUE THREAD

**Issue by Rafkraft** [balderdashy/skipper#106]
Hi

I'm experiencing an issue which has been largely discussed on multiple platforms, but I didn't find any clear answer. I'm uploading a simple image file thru a basic form. In my controller, I pre-test multiple attributes of that file (content-type, unicity, presence). When these tests are passed, the file is uploaded on the server without problems, but when it fails, the file remains in the buffer, and the server crashes few seconds later because the file has not been plugged in the skipper receiver. I get the following error:

```
events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: EMAXBUFFER: An Upstream (`NOOP_logo`) timed out before it was plugged into a receiver. It was still unused after waiting 4500ms. You can configure this timeout by changing the `maxTimeToBuffer` option.
    at null.<anonymous> (PATH_TO_PROJECT/node_modules/sails/node_modules/skipper/standalone/Upstream/Upstream.js:86:15)
    at Timer.listOnTimeout [as ontimeout] (timers.js:110:15)
```

One dirty way to handle it is, in case of pre-test fail, to upload the file to a "trash" directory, and delete it afterwards, but isn't there a simpler way to delete the file / clear buffer ???
NOTE: the reqFile.upload(function() {}); to clear the buffer doesn't work. Same error is raised.

Question 2: isn't there a way to check file's content-type using skipper's upload() method ???

My controller code:
```
  logo:function(req, res){
    
    if (!req.file('logo')){
      // Server will crash in few seconds
      req.addFlash('error', req.__("err1263"));
      return res.redirect('account/edit')
    }

    var file = req.file('logo')._files[0].stream;
    var contentType = file.headers['content-type'];
    if(!_.contains(['image/jpeg', 'image/jpg', 'image/png'], contentType)){
      // Server will crash in few seconds
      req.addFlash('error', req.__("err1263"));
      return res.redirect('account/edit')
    }

    var uploadPath = sails.config.appPath + '/public/logos/';
    req.file(name).upload({
      dirname: uploadPath,
      saveAs: 'logo_name',
      maxBytes: 1000000
    }, function (err, file) {

      if(err && err.code == 'E_EXCEEDS_UPLOAD_LIMIT'){
        // Server will not crash because file has been handled
        req.addFlash('error', req.__("err1264"));
        return res.redirect(redirectUrl)
      }

      req.addFlash('error', req.__("inf2000"));
      return res.redirect(redirectUrl)
    });
  }
```
My view (jade):
```
form(method="POST", action="/account/logo", enctype="multipart/form-data")
  .form-group
    input(type="file", name="logo")
  .form-group
    button.btn.btn-primary(type='submit') Edit logo
```

Thank you very much, I've been stuck with this issue for weeks.

**Comment by sailsbot** [comment]
Thanks for posting, @Rafkraft.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by ebbnormal** [comment]
@Rafkraft  did you ever figure out this error? I am running into the same issue!

**Comment by Rafkraft** [comment]
nope :( pm me if you find somethig

**Comment by ebbnormal** [comment]
@Rafkraft What was happening, in my case, was that I didn't set the `<form>` `enctype` content type to ` `multipart/form-data`. So the POST request dropped my files, and thus I got the Buffer error, as no data was sent for Sails to parse, in terms of `req.files`.
